pkg/azurefile/controllerserver.go

```diff
@@ -311,25 +350,51 @@ func (d *Driver) getShareUrl(sourceVolumeID string) (azfile.ShareURL, error) {
 
 	accountKey, err := d.cloud.GetStorageAccesskey(accountName, resourceGroupName)
 	if err != nil {
-		return azfile.ShareURL{}, fmt.Errorf("could not find key for storage account(%s) under resource group(%s), err %v", accountName, resourceGroupName, err)
+		return azfile.ServiceURL{}, "", fmt.Errorf("could not find key for storage account(%s) under resource group(%s), err %v", accountName, resourceGroupName, err)
 	}
 
 	credential, err := azfile.NewSharedKeyCredential(accountName, accountKey)
 	if err != nil {
 		klog.Errorf("NewSharedKeyCredential(%s) in CreateSnapshot failed with error: %v", accountName, err)
-		return azfile.ShareURL{}, err
+		return azfile.ServiceURL{}, "", err
 	}
 
 	u, err := url.Parse(fmt.Sprintf(fileUrlTemplate, accountName, d.cloud.Environment.StorageEndpointSuffix))
 	if err != nil {
 		klog.Errorf("parse fileUrlTemplate error: %v", err)
-		return azfile.ShareURL{}, err
+		return azfile.ServiceURL{}, "", err
 	}
 	if u == nil {
-		return azfile.ShareURL{}, fmt.Errorf("url is nil")
+		return azfile.ServiceURL{}, "", fmt.Errorf("url is nil")
 	}
 
 	serviceURL := azfile.NewServiceURL(*u, azfile.NewPipeline(credential, azfile.PipelineOptions{}))
 
-	return serviceURL.NewShareURL(fileShareName), nil
+	return serviceURL, fileShareName, nil
+}
+
+func (d *Driver) snapshotExists(ctx context.Context, sourceVolumeID string, name string) (bool, azfile.ShareItem, error) {
```

Comment:
should explain `sourceVolumeID`, and `name` should be `snapshotName`?

Reply:
done.